pkg/cloudprovider/providers/gce/gce_util.go

```diff
@@ -211,3 +219,58 @@ func handleAlphaNetworkTierGetError(err error) (string, error) {
 	// Can't get the network tier, just return an error.
 	return "", err
 }
+
+// containsCIDR returns true if outer contains inner.
+func containsCIDR(outer, inner *net.IPNet) bool {
+	return outer.Contains(firstIPInRange(inner)) && outer.Contains(lastIPInRange(inner))
+}
+
+// firstIPInRange returns the first IP in a given IP range.
+func firstIPInRange(ipNet *net.IPNet) net.IP {
+	return ipNet.IP.Mask(ipNet.Mask)
+}
+
+// lastIPInRange returns the last IP in a given IP range.
+func lastIPInRange(cidr *net.IPNet) net.IP {
+	ip := append([]byte{}, cidr.IP...)
+	for i, b := range cidr.Mask {
+		ip[i] |= ^b
+	}
+	return ip
+}
+
+// subnetsInAutoCIDR takes a list of subnets for a single region and
+// returns subnets which exists in the auto CIDR range.
+func subnetsInAutoCIDR(subnets []*compute.Subnetwork) ([]*compute.Subnetwork, error) {
+	var res []*compute.Subnetwork
```

Comment:
I would suggest changing the signature of the function to take in a CIDR range and change the function name to something like subnetsInCIDR(). This would make the function more generic and fix the issue we discussed offline.

Reply:
Good call, thanks!